### ISSUE TITLE
SEED procedure not correctly inserting user data with Identity/SQL Server

### ISSUE THREAD

**Issue by LeadMagnet** [PiranhaCMS/piranha.core.docs#27]
If I create a new project using the "dotnet new" statements on the main page, the user credentials are inserted into the SQLLite (default) database and I can log into the manager.

Create another project, this time omitting the "dotnet run" command. Then following the documentation change the config/code to use SQL Server and Identity(with seed). Compile the project and run (in VS).

The database is correctly populated with objects. The data to display the default pages is inserted. All dbo.Piranha_User* tables are blank.

Going to the manager page, using "admin / password" gets an "incorrect username/password" error message. Which (of course) happens because these tables are empty.

I've tried to find the "seed" procedure that inserts SQL Server data so I can manually run it, but I cannot find it.

Can you please post the SQL INSERT commands I need to successfully log in?

**Comment by tidyui** [comment]
Hi @LeadMagnet This usually happens if you change from `SQLite` to `SQLServer` without changing database-provider. The reason for this is that Identity uses auto-increment columns that are handled differently between. Take a look at this doc page:

https://piranhacms.org/docs/architecture/authentication/identity

Basically you will need to install the package:

    Piranha.AspNetCore.Identity.SQLServer

Then you need to change the service registration so that you're using the correct `DbContext`.

    services.AddPiranhaIdentityWithSeed<IdentitySQLServerDb>(options => ...);

**Comment by LeadMagnet** [comment]
Yes I did this, BEFORE running the app for the first time. Just before running, I deleted the database to be 100% sure the setup and seed procedures would fire.

Still no user data.

If a create a fresh project using the default SQLLite, the user data is inserted just fine.

Do you have the insert commands to manually insert user data? Or at least pointme to the seed function which _**should**_ be firing?

**Comment by tidyui** [comment]
The default `SysAdmin` role is created and attached to all available Claims here:

https://github.com/PiranhaCMS/piranha.core/blob/master/core/Piranha.AspNetCore.Identity/Db.cs#L93

The default user is seeded here:

https://github.com/PiranhaCMS/piranha.core/blob/master/core/Piranha.AspNetCore.Identity/DefaultIdentitySeed.cs#L50

Best regards

**Comment by tidyui** [comment]
Which version are using so we can see if we can recreate your issue?

**Comment by tidyui** [comment]
And I'm transferring this is to the `piranha.core` repo.
